### DISCUSSION TITLE
abs-tract: A custom metadata provider for Goodreads and Kindle

### DISCUSSION THREAD

**Discussion by ahobsonsayers** [advplyr/audiobookshelf#2890] [Show and tell]
Hi all!

I wanted to share a custom metadata provider I have built for Audiobookshelf and I thought this would be the appropriate place to post it!

https://github.com/ahobsonsayers/abs-tract

The provider currently pulls metadata from Goodreads and Kindle and is more focused on ebooks than audiobooks, although both will work.

In my opinion, Goodreads is unbeatable for quality and complete ebook metadata, while Kindle provides the best quality covers for books from any source (seriously, the files are huge). These providers used together have significantly improved my library with better metadata and better-looking covers, and I really hope they will be able to help you as well.

As far as I can tell, this is the first custom metadata provider built for Audiobookshelf available on GitHub, using the recent addition of custom metadata providers (please let me know if this is not correct; I'd love to see others!). As such, please give abs-tract a try and let me know what you think! Please also open issues or contribute if you have any ideas or feedback!

**Comment by nichwall** [comment]
Looks nice! Feel free to add the provider to the "community provider" table if you'd like.

https://www.audiobookshelf.org/guides/custom-metadata-providers

**Reply by ahobsonsayers** [reply]
Thanks for the suggestion! I have opened a PR here: https://github.com/audiobookshelf/audiobookshelf-web/pull/90
